mautic/mautic
# Check for 'select' type fields on import

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ Yes]
| New feature/enhancement? (use the a.x branch)      | [ No]
| Deprecations?                          | [ NA]
| BC breaks? (use the c.x branch)        | [ NA]
| Automated tests included?              | [Yes ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:



This PR ensures that no gibberish value is accepted for a 'select type' lead field field while importing contacts from a csv. This is achieved by adding a check for the same while saving the entity.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a custom field 'gender' with following labels: Male, Female and options: male, female respectively
3. Create a CSV with a gender column having values like 'male', 'female, 'Male', 'Female', 'gibberish', 'yahoo', etc. Refer to the CSV below if needed. This is the one I used to test this code.
[gender2.csv](https://github.com/acquia/mc-cs/files/13377348/gender2.csv)

5. Import this CSV 
6. Check `mautic_leads` table's gender column. This column should only have 'male' and 'female' values. All the rest of the contacts with gibberish/disallowed values should have `null` in the gender column. 

#### Other areas of Mautic that may be affected by the change:
1. Any area which makes a call to [LeadModel::saveEntity()](https://github.com/mautic/mautic/blob/18ebaec0309fe208f1d76e175675fc62e211557b/app/bundles/LeadBundle/Model/LeadModel.php#L366), basically all ways we can create and save a contact.

[//]: # ( As applicable: )
#### List of areas covered by the unit and/or functional tests:
1. Creating a contact and adding the allowed and disallowed values of a 'select' type custom field to it.
2.
